tmVhdlProducer/main.py

```diff
@@ -184,6 +186,17 @@ def main() -> int:
                 raise ValueError(f"no such constraint: {alias!r}")
             for key in ConstraintTypes[alias]:
                 constraints[key] = module
+
+    # Default constraint for ZDC
+    zdc_constraint = True
+    for key in constraints.keys():
+        if (key == kZDCPlus) or (key == kZDCMinus):
+            zdc_constraint = True
+        else:
+            zdc_constraint = False
+    if (not zdc_constraint) or (len(constraints) == 0):
+        constraints.update( [(kZDCPlus, [DefaultModuleZdc]),(kZDCMinus, [DefaultModuleZdc])] )
+            
```

Comment:
That is a bit complicated, how about this:
```python
# Default constraint for ZDC
constraints.setdefault(kZDCMinus, [DefaultModuleZdc])
constraints.setdefault(kZDCPlus, [DefaultModuleZdc])
```
Also using [list comprehensions](https://realpython.com/list-comprehension-python/#how-to-supercharge-your-comprehensions) would work:
```python
# Default constraint for ZDC
if kZDCMinus not in constraints:
    constraints[kZDCMinus] = [DefaultModuleZdc]
if kZDCPlus not in constraints:
    constraints[kZDCPlus] = [DefaultModuleZdc]
```